v2/api/storage/v1beta20210401/storage_accounts_management_policy_types_gen.go

```diff
@@ -562,7 +562,7 @@ func (policies *StorageAccountsManagementPolicies_Spec) ConvertToARM(resolved ge
 		if err != nil {
 			return nil, err
 		}
-		policy := policyARM.(ManagementPolicySchemaARM)
+		policy := *policyARM.(*ManagementPolicySchemaARM)
 		result.Properties.Policy = &policy
```

Comment:
Trying to understand the purpose of this - is it to ensure that the original and the ARM variant are completely independent in memory?

Reply:
No it's an artefact of the code generation. I describe the exact "why" of this in the summary of this PR, but the tl;dr is that this code block is generated by a set of handlers calling themselves recursively. So there's two handlers at play here, the inner handler, whose contract is to return a non-ptr, and an outer handler which is the `Optional` handler whose contract is to return a pointer.

In this actual code, this bit is generated by the inner handler:
```
	policyARM, err := (*policies.Policy).ConvertToARM(resolved)
	if err != nil {
		return nil, err
	}
	policy := policyARM.(ManagementPolicySchemaARM)
	policy := *policyARM.(*ManagementPolicySchemaARM) // Result is a non-ptr, as per contract
```

and this bit is generated by the outer (optional) handler:
```
	if policies.Policy != nil {
		<inner>
		result.Properties.Policy = &<inner result> // result is a ptr
	}
```

The only way I see to fix this is that we change the target type the inner handler is trying to "get" to to be a ptr rather than a struct. This means delegating the `&<result>` to that inner handler. The problem is there's a lot of possible inner handlers that can get called which means I need to increase each of their complexity to deal with this case. I can't see an obvious clean way to make sure to generate code that is more human-like that doesn't also increase the complexity of this already complicated code, so I punted on it for now, favoring readability of the generator over readability of the generated code.

It'd be nice to have both though.